tox.ini

```diff
@@ -1,50 +1,14 @@
 [tox]
-envlist = py{37,38,39}
+envlist = py{38,39,310,311,312}
 
 [testenv]
-whitelist_externals = echo make
 deps =
     -rrequirements.txt
-    flake8
 allowlist_externals =
     echo
     make
-commands =
-    # $EXCLUDE is defined above in testenv:py27 as a workaround for Python 2
-    # which does not support asyncio and type hints
-    flake8 . --count --select=E9,F63,F7,F822,F823 --show-source --statistics {env:EXCLUDE:}
-    python setup.py test
-    make -C test test
-    # TODO (cclauss) Fix up all these undefined names
-    flake8 . --count --exit-zero --select=F821 --show-source --statistics
-
-# On older Python, flake8 version 4 fail with:
-# RecursionError: maximum recursion depth exceeded
-[testenv:py37]
-deps =
-    -rrequirements.txt
-    flake8<4
-
-# This lint environment should be the same as the one in .github/work
-[testenv:lint]
```

Comment:
Kept the lint environment, this allow to run linter locally. I want to be able to run lints before pushing to github.

The idea is the the tox contains the same linter that are currently present in github workflow, so locally or Github could call `tox -e lint` to check linters.

I've made few change with what was run by Github lint action, mostly instead of `.` I use `src`. This was because without it, linters would process the `.tox` folder which is quiet large.